bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaValues.java

```diff
@@ -79,14 +81,46 @@
     public static ObjectValue createObjectValue(BPackage packageId, String objectTypeName, Object... fieldValues) {
         ValueCreator valueCreator = ValueCreator.getValueCreator(packageId.toString());
         Object[] fields = new Object[fieldValues.length * 2];
+        Strand currentStrand = getStrand();
+        // Adding boolean values for each arg
+        for (int i = 0, j = 0; i < fieldValues.length; i++) {
+            fields[j++] = fieldValues[i];
+            fields[j++] = true;
+        }
+        return valueCreator.createObjectValue(objectTypeName, getScheduler(currentStrand), currentStrand, null, fields);
+    }
+
+    public static ObjectValue createObjectValue(BPackage packageId, String objectTypeName, Strand currentStrand,
+                                                Object... fieldValues) {
+        //This method duplicates the createObjectValue with referencing the issue in runtime API getting strand
+        ValueCreator valueCreator = ValueCreator.getValueCreator(packageId.toString());
+        Object[] fields = new Object[fieldValues.length * 2];
 
         // Adding boolean values for each arg
         for (int i = 0, j = 0; i < fieldValues.length; i++) {
             fields[j++] = fieldValues[i];
             fields[j++] = true;
         }
-        //passing scheduler, strand and properties as null for the moment, but better to expose them via this method
-        return valueCreator.createObjectValue(objectTypeName, null, null, null, fields);
+        //passing scheduler of current strand
+        return valueCreator.createObjectValue(objectTypeName, currentStrand.scheduler, currentStrand,
+                null, fields);
+    }
+
+    private static Scheduler getScheduler(Strand currentStrand) {
+        try {
+            return currentStrand.scheduler;
+        } catch (Exception ex) {
+            return null;
```

Comment:
when can an exception occur here?

Reply:
Null pointer exception is thrown if the strand is null.